src/main/java/com/redhat/devtools/lsp4ij/features/workspaceSymbol/AbstractLSPWorkspaceSymbolContributor.java

```diff
@@ -1,109 +1,127 @@
+import com.intellij.util.indexing.IdFilter;
+import com.redhat.devtools.lsp4ij.LSPWorkspaceSupport;
+import org.eclipse.lsp4j.WorkspaceSymbolParams;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import static com.redhat.devtools.lsp4ij.internal.CompletableFutures.isDoneNormally;
+import static com.redhat.devtools.lsp4ij.internal.CompletableFutures.waitUntilDone;
+
+/**
+ * Abstract base class for LSP goto symbol contributors
+ */
+abstract class AbstractLSPWorkspaceSymbolContributor implements ChooseByNameContributorEx {
+
+    private final static Logger LOGGER = LoggerFactory.getLogger(AbstractLSPWorkspaceSymbolContributor.class);
+
+    @Override
+    public void processNames(@NotNull Processor<? super String> processor,
+                             @NotNull GlobalSearchScope scope,
+                             @Nullable IdFilter filter) {
+        Project project = scope.getProject();
+        if (project == null) {
+            return;
+        }
+        String queryString = project.getUserData(ChooseByNamePopup.CURRENT_SEARCH_PATTERN);
+        if (queryString == null) {
+            queryString = "";
+        }
+        List<WorkspaceSymbolData> items = getWorkspaceSymbols(queryString, true, project);
+        if (items != null) {
+            items.stream()
+                    .filter(data -> scope.accept(data.getFile()))
+                    .map(NavigationItem::getName)
+                    .forEach(processor::process);
+        }
+    }
+
+    @Override
+    public void processElementsWithName(@NotNull String name,
+                                        @NotNull Processor<? super NavigationItem> processor,
+                                        @NotNull FindSymbolParameters parameters) {
+        List<WorkspaceSymbolData> items = getWorkspaceSymbols(name, false, parameters.getProject());
+        if (items != null) {
+            items
+                    .stream()
+                    .filter(ni -> parameters.getSearchScope().accept(ni.getFile()))
+                    .forEach(processor::process);
+        }
+    }
+
+    private List<WorkspaceSymbolData> getWorkspaceSymbols(@NotNull String name, boolean cancel, Project project) {
+        // Consume LSP 'workspace/symbol' request
+        LSPWorkspaceSymbolSupport workspaceSymbolSupport = LSPWorkspaceSupport.getSupport(project).getWorkspaceSymbolSupport();
+        if (cancel) {
+            workspaceSymbolSupport.cancel();
+        }
+        WorkspaceSymbolParams params = new WorkspaceSymbolParams(name);
+        CompletableFuture<List<WorkspaceSymbolData>> workspaceSymbolFuture = workspaceSymbolSupport.getWorkspaceSymbol(params);
```

Comment:
An annoying problem that I see (with our IJ Quarkus) is that our workspace symbol doesn't manage Class, Struct, etc and  our workspace symbol can takes some times to collect workspace symbol which scans all classpath to collect REST services in Java.

In Java, the Classes search is a lot of used and it means that we will execute the workspace symbol from our IJ Quarkus although we are sure that it will not report some Classes and it will impact performances of Class search.

To avoid this problem, it would be nice to add a new method like `LSPWorkspaceSymbolFeature#canReportClasses()` (@fbricon what do you think about the name) which will return true if the workspace symbol must be executed when user will search Classes and false otherwise.

By default it should return false. For user defined language server definition, we could add a new settings (same settings that caseSensitive completion like you want to add) to know if language server can report classes by executing workspace symbol.

Reply:
@angelozerr, I would propose that the default for such a setting should be true as the potential performance issue you've described seems to be the exception and most language servers could presumably respond quickly.

Reply:
I think we should set to false by default because I fear about workspace symbol performance from language servers.

And for user defined language servers we should use a settings, but if you want to do that in an another PR, you could set to true for UserDefinedLanguageServers for the moment.